### PR TITLE
feat: add support for NUMERIC data type

### PR DESCRIPTION
**Pull request by olavloite**

### CODE REVIEW

File: google-cloud-spanner/src/test/java/com/google/cloud/spanner/connection/RandomResultSetGenerator.java [hunk 1]
```diff
@@ -41,6 +42,7 @@
         Type.newBuilder().setCode(TypeCode.BOOL).build(),
         Type.newBuilder().setCode(TypeCode.INT64).build(),
         Type.newBuilder().setCode(TypeCode.FLOAT64).build(),
+        //        Type.newBuilder().setCode(TypeCode.NUMERIC).build(),
```
Comment (thiagotnunes) [inline google-cloud-spanner/src/test/java/com/google/cloud/spanner/connection/RandomResultSetGenerator.java:45]: nit: Commented out code. Maybe we forgot to remove it?

Comment (olavloite) [reply google-cloud-spanner/src/test/java/com/google/cloud/spanner/connection/RandomResultSetGenerator.java]: Ah, good catch. This is not supposed be commented out at all.

File: google-cloud-spanner/src/test/java/com/google/cloud/spanner/it/ITWriteTest.java [hunk 2]
```diff
@@ -352,6 +356,21 @@ public void writeDateNull() {
     assertThat(row.isNull(0)).isTrue();
   }
 
+  @Test
```
Comment (skuruppu) [inline google-cloud-spanner/src/test/java/com/google/cloud/spanner/it/ITWriteTest.java:359]: @olavloite thanks for adding the integration tests. But I do have one small request. Would you be able to move it to a separate PR?

The reason being that we will merge this PR in before support for the feature is available in prod, which unfortunately mean that the tests will fail at that point.

Comment (olavloite) [reply google-cloud-spanner/src/test/java/com/google/cloud/spanner/it/ITWriteTest.java:359]: NP, I'll move them to a separate PR.
